.github/workflows/qodana.yml

```diff
@@ -6,22 +6,20 @@ on:
   push:
     branches:
       - master
+
 jobs:
-  inspection:
+  qodana:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - name: Qodana - Code Inspection
-        id: qodana
-        uses: JetBrains/qodana-action@v2022.2.1
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: 'Qodana Scan'
+        uses: JetBrains/qodana-action@v2022.2.3
         with:
           args: --baseline,qodana.sarif.json,--fail-threshold,0,--linter,jetbrains/qodana-jvm
         env:
           QODANA_TOKEN: ${{ secrets.QODANA_TOKEN }}
-  clone-finder:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - name: Qodana - Clone Finder
-        uses: JetBrains/qodana-clone-finder-action@v2.0-eap
+      - uses: github/codeql-action/upload-sarif@v2
+        with:
+          sarif_file: ${{ runner.temp }}/qodana/results/qodana.sarif.json
```

Comment:
Not sure how it works and what it'll look like, but it's configuration for [GitHub code scanning](https://github.com/JetBrains/qodana-action#github-code-scanning), which sounds nice